test/e2e/e2e.go

```diff
@@ -40,3 +45,73 @@ func SetupWithNamespace(t *testing.T, namespace string) *test.Clients {
 	}
 	return clients
 }
+
+// CreateNamespace creates a namespace to run the integration tests.
+func CreateNamespace(t *testing.T, clients *test.Clients, namespace string) {
```

Comment:
Please do not do this in code. It prevents you from modifying the namespaces in third-party integration systems should you need it. It's also different from what knative-serving does, where the namespaces are created from `test/config`

Reply:
Another benefit, at least of not having a randomized namespace name, is that once we have subtests we can run them individually, e.g.
```
[jim@t450s serving-operator]$ go test -run=/create -v -tags=e2e -count=1 ./test/e2e
=== RUN   TestKnativeServingDeployment
=== RUN   TestKnativeServingDeployment/create
--- PASS: TestKnativeServingDeployment (50.12s)
    --- PASS: TestKnativeServingDeployment/create (50.02s)
PASS
ok  	knative.dev/serving-operator/test/e2e	50.154s
[jim@t450s serving-operator]$ go test -run=/delete -v -tags=e2e -count=1 ./test/e2e
=== RUN   TestKnativeServingDeployment
=== RUN   TestKnativeServingDeployment/delete
--- FAIL: TestKnativeServingDeployment (0.09s)
    --- FAIL: TestKnativeServingDeployment/delete (0.02s)
        e2e.go:105: KnativeService "knative-serving" failed to delete: knativeservings.serving.knative.dev "knative-serving" not found
FAIL
FAIL	knative.dev/serving-operator/test/e2e	0.122s
```
It's the randomized namespace that causes the 2nd failure.